app/test/ica_test.go

```diff
@@ -0,0 +1,116 @@
+package app_test
+
+import (
+	"encoding/json"
+	"testing"
+	"time"
+
+	"github.com/celestiaorg/celestia-app/app"
+	"github.com/celestiaorg/celestia-app/app/encoding"
+	v1 "github.com/celestiaorg/celestia-app/pkg/appconsts/v1"
+	v2 "github.com/celestiaorg/celestia-app/pkg/appconsts/v2"
+	"github.com/celestiaorg/celestia-app/test/util"
+	"github.com/cosmos/cosmos-sdk/crypto/keyring"
+	"github.com/cosmos/cosmos-sdk/x/params/types/proposal"
+	icahosttypes "github.com/cosmos/ibc-go/v6/modules/apps/27-interchain-accounts/host/types"
+	"github.com/stretchr/testify/require"
+	abci "github.com/tendermint/tendermint/abci/types"
+	"github.com/tendermint/tendermint/libs/log"
+	tmproto "github.com/tendermint/tendermint/proto/tendermint/types"
+	version "github.com/tendermint/tendermint/proto/tendermint/version"
+	dbm "github.com/tendermint/tm-db"
+)
+
+// TestICA verifies that the ICA module's params are overridden during an
+// upgrade from v1 -> v2.
+func TestICA(t *testing.T) {
+	testApp, _ := setupTestApp(t, 3)
+	supportedVersions := []uint64{v1.Version, v2.Version}
+	require.Equal(t, supportedVersions, testApp.SupportedVersions())
+
+	ctx := testApp.NewContext(true, tmproto.Header{
+		Version: version.Consensus{
+			App: 1,
+		},
+	})
+	testApp.BeginBlock(abci.RequestBeginBlock{Header: tmproto.Header{
+		Height:  2,
+		Version: version.Consensus{App: 1},
+	}})
+
+	// app version should not have changed yet
+	require.EqualValues(t, 1, testApp.AppVersion())
+
+	// global min gas price should not have been set yet
+	gotBefore, err := testApp.ParamsKeeper.Params(ctx, &proposal.QueryParamsRequest{
+		Subspace: icahosttypes.SubModuleName,
+		Key:      string(icahosttypes.KeyHostEnabled),
+	})
+	require.Equal(t, "", gotBefore.Param.Value)
+	require.NoError(t, err)
+
+	// now the app version changes
+	respEndBlock := testApp.EndBlock(abci.RequestEndBlock{Height: 2})
+	testApp.Commit()
+
+	require.NotNil(t, respEndBlock.ConsensusParamUpdates.Version)
+	require.EqualValues(t, 2, respEndBlock.ConsensusParamUpdates.Version.AppVersion)
+	require.EqualValues(t, 2, testApp.AppVersion())
+
+	// create a new context after endBlock
+	newCtx := testApp.NewContext(true, tmproto.Header{
+		Version: version.Consensus{
+			App: 2,
+		},
+	})
+
+	got, err := testApp.ParamsKeeper.Params(newCtx, &proposal.QueryParamsRequest{
+		Subspace: icahosttypes.SubModuleName,
+		Key:      string(icahosttypes.KeyHostEnabled),
+	})
```

Comment:
nit: Does the ICA have it's own grpc method for getting the params? It's probably better to query that (as it should set the context for you and return all params, not just the `HostEnabled` one

Reply:
ICA does have a GRPC method for params:

<img width="582" alt="Screenshot 2024-04-02 at 1 53 58 PM" src="https://github.com/celestiaorg/celestia-app/assets/3699047/386fef0b-506f-47ab-9176-326236b98bf0">

I just can't find code that queries params via GRPC from the testApp. Will attempt to switch from ParamsKeeper to GRPC method

Reply:
Done